examples/nemo/scripts/omp_gpu_trans.py

```diff
@@ -112,6 +114,25 @@ def trans(psy):
                     OMPDeclareTargetTrans().apply(invoke.schedule)
                     continue
 
+        # For now this is a special case for stpctl.f90 because it forces
+        # loops to parallelise without many safety checks
+        if psy.name == "psy_stpctl_psy":
+            for loop in invoke.schedule.walk(Loop):
+                # Skip if an outer loop is already parallelised
+                if loop.ancestor(Directive):
+                    continue
+                omp_loop_trans.apply(loop, options={"force": True})
+                omp_target_trans.apply(loop.parent.parent)
+                assigns = loop.walk(Assignment)
+                if len(assigns) == 1 and assigns[0].lhs.symbol.name == "zmax":
+                    stmt = assigns[0]
+                    if OMPAtomicDirective.is_valid_atomic_statement(stmt):
```

Comment:
This should really be done with a Transformation. For now I'll accept a TODO though (unless there's some issue with having a transformation - I very vaguely remember that there might be?).

Reply:
I created #2446 and added the TODO. The problem is that the loop_trans needs to be aware of the atomic (same with reductions) otherwise it can not guarantee that it is valid (this is why for this file I pass the 'force' option). I think the best solution is to do both from inside the loop_transformation but I prefer to discuss this in the new issue.